### PR TITLE
Update to sebastian/diff 2

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/sebastianbergmann/comparator/issues/37

### CODE REVIEW

File: .php_cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+$header = <<<'EOF'
+This file is part of the Comparator package.
+
+(c) Sebastian Bergmann <sebastian@phpunit.de>
+
+For the full copyright and license information, please view the LICENSE
+file that was distributed with this source code.
+EOF;
+
+return PhpCsFixer\Config::create()
```
Comment (SpacePossum) [inline .php_cs]: copy of the PHPUnit one with few rules removed

**Comment by SpacePossum** [comment]
We need this before https://github.com/sebastianbergmann/phpunit/issues/2734 can be done, as this package doesn't allow diff 2.0 (for obvious reasons :)).

**Comment by sebastianbergmann** [comment]
Please send separate pull requests for code style changes etc. and the actual update to support the new version of the diff component (maybe #36 is enough?).

**Comment by sebastianbergmann** [comment]
I have merged #36.

**Comment by SpacePossum** [comment]
36 might get what you want, however kindly note that nothing of it is under tests.
Secondly it will break for others as the dependency is not raised to minimum of `diff v2` within `composer.json` in that PR.
I could tree shake this PR a bit, but in the end most of it is either about the upgrade and making sure the upgrade is covered by tests. 
Not sure why this PR was dropped in fav. of one not providing tests and only updates half of what is needed, v.s. one that brings +10% coverage and both the needed updates. Now I've resolve conflicts and reopen the PR because the time between RFC on it and the closing was 6 minutes?
I hope to find the time tomorrow to rework this, but yes maybe it is not needed at all?

**Comment by sebastianbergmann** [comment]
I am sorry if I caused you trouble, that was not my intention.

What confuses me, and I understand that you did not want to confuse me intentionally, is to mix separate concerns in one pull request.

I had not seen #36 before. It looked sufficient at first glance so I merged it. I then updated `composer.json` myself and tagged the `2.0.1` release.

I plan to work on https://github.com/sebastianbergmann/phpunit/issues/2734 later tonight.

**Comment by SpacePossum** [comment]
No worries, all good :)
I overlooked 36 as well TBH. I'll split up this PR to see if there is anything left of interest here.

**Pull request closed without merging**